refactor(persistQueryClient): Make persistQueryClient stable

prateek3255 opened the pull request:
- Rename Persistor -> Persister
- Remove experimental from persistQueryClient, createWebStoragePersister and createAsyncStoragePersister.

Fixes #2921

prateek3255 commented:
@TkDodo Should I update the docs as well?

TkDodo commented:
> @TkDodo Should I update the docs as well?

yes please. And also, we have a migration to v4 guide, where it would be good to mention that this is now stable

TkDodo commented on package.json:
```diff
@@ -55,7 +55,7 @@
     "devtools",
-    "persistQueryClient-experimental",
+    "persistQueryClient",
     "createWebStoragePersistor-experimental",
     "createAsyncStoragePersistor-experimental",
```
those two should become stable as well please, as they go hand in hand with `persistQueryClient`.

and also renamed to `persister` please :)

prateek3255 commented:
@TkDodo This is GTG from my side now, have renamed all the plugins, updated the docs, and also added this change to the migration guide. I have also [created a sandbox](https://codesandbox.io/s/tannerlinsley-react-query-basic-typescript-forked-6j4z7) that uses the build from this PR, with these renamed plugins if you want to test it out.

TkDodo commented on docs/src/pages/guides/migrating-to-react-query-4.md:
```diff
@@ -132,6 +132,22 @@ If you were importing anything from `'react-query/react'` directly in your proje
+
+The plugins `createWebStoragePersistor` and `createAsyncStoragePersistor` have been renamed to [`createWebStoragePersister`](/plugins/createWebStoragePersister) and [`createAsyncStoragePersister`](/plugins/createAsyncStoragePersister) respectively. The interface `Persistor` in `persistQueryClient` has also been renamed to `Persister`. Checkout [this stackexchange](https://english.stackexchange.co ...
+
+Since these plugins are no longer experimental, there import paths have also been updated:
```
```suggestion
Since these plugins are no longer experimental, their import paths have also been updated:
```

TkDodo reviewed (approved):
really good, I'll just fix the two typos and merge it, thank you for working on this 🚀
